src/main/java/org/akip/service/NotifyUserAssignedTasksService.java

```diff
@@ -0,0 +1,57 @@
+package org.akip.service;
+
+import org.akip.domain.TaskInstance;
+import org.akip.domain.enumeration.StatusTaskInstance;
+import org.akip.repository.TaskInstanceRepository;
+import org.akip.resolver.AkipUserDTO;
+import org.akip.resolver.UserResolver;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+@Service
+@Transactional
+public class NotifyUserAssignedTasksService {
+
+    private final AkipMailService mailService;
+
+    private final TaskInstanceRepository taskInstanceRepository;
+
+    private final UserResolver userResolver;
+
+    public NotifyUserAssignedTasksService(AkipMailService mailService, TaskInstanceRepository taskInstanceRepository, UserResolver userResolver) {
+        this.mailService = mailService;
+        this.taskInstanceRepository = taskInstanceRepository;
+        this.userResolver = userResolver;
+    }
+
+    public Map<String, String> notifyUserAssignedTasks(){
+        Map<String, String> summary = new HashMap<>();
+        Map<String, Object> variables = new HashMap<>();
+
+        List<TaskInstance> tasksAssigned = taskInstanceRepository.findByStatus(StatusTaskInstance.ASSIGNED);
+
+        Set<String> usersLogin = tasksAssigned
+            .stream()
+            .map(TaskInstance::getAssignee)
+            .collect(Collectors.toSet());
+
+        List<String> userLoginList = new ArrayList<>(usersLogin);
+
+        List<AkipUserDTO> users = userResolver.getUsersByLogins(userLoginList);
+
+        for (AkipUserDTO user : users){
+            List<TaskInstance> userTasks = tasksAssigned
+                .stream()
+                .filter((task) -> task.getAssignee().equals(user.getLogin()))
+                .collect(Collectors.toList());
+
+            variables.put("tasks", userTasks);
+            mailService.sendNotifyUserAssignedTasksMail(user, variables);
+            summary.put("sentEmailsTo", "Sent Emails To "+user.getEmail()+": " + userTasks.size());
```

Comment:
The idea of the summary map is to use a key/summary structure. So, it is not making sense to use the same key "sentEmailsTo" for iterations. As you are using the same key, the resulting map will have only one element: the last one included.